### ISSUE TITLE
[question] How to do column highlighting / selection? 

### ISSUE THREAD

**Issue by alfonsoar** [mui/mui-x#6397]
### Order ID 💳

2022-14565

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

I am looking for examples on how to do column highlighting/selection but for columns. Similar to how we can do [row selection ](https://mui.com/x/react-data-grid/selection/#row-selection) I am looking to implement a version of this but for columns. 

Ideally I would want to find something that is native but would also be willing to implement something on my end if it does not exists. 

For reference ag-grid does column highlighting like this: https://www.ag-grid.com/archive/21.2.2/javascript-grid-highlighting-rows-and-columns/

**Comment by alfonsoar** [comment]
I was able to get something working but unsure if we have a better way: https://stackblitz.com/edit/react-sr8mci?file=demo.tsx

**Comment by alfonsoar** [comment]
Found a better way to do it: https://stackblitz.com/edit/react-wzrysy?file=demo.tsx

**Comment by yaredtsy** [comment]
@alfonsoar that's a useful feature. there are an `onColumnHeaderLeave` and `onColumnHeaderEnter` API you can use them for the hover effect https://stackblitz.com/edit/react-wzrysy-6qjc9n?file=demo.tsx

**Comment by alfonsoar** [comment]
@yaredtsy nice, should we include this in the docs? I can PR it if yes

**Comment by yaredtsy** [comment]
I think it will be better if it was a built-in feature. 

@DanailH  should we include this in the docs?

**Comment by DanailH** [comment]
Well, it's an interesting case but I'm more concerned about the user expectation. If it behaves visually as the row selection then users might expect that when they highlight a column that they will get the same features (copy/past, clipboard export, etc) as when rows are selected.

**Comment by yaredtsy** [comment]
it can be optional, users can enable it when they want it.  
for example, column spanning can be useful, to check which cell belongs to which column. here is an example https://codesandbox.io/s/gallant-keller-8s3unr?file=/demo.js

**Comment by DanailH** [comment]
Yes but this is only visible while you hover over the column. When you click it nothing happens (except sorting).
Either way, I would like to the opinions of @joserodolfofreitas and @gerdadesign regarding this.

**Comment by joserodolfofreitas** [comment]
@alfonsoar, I'm curious to understand more about your use case.
Could you expand on why you are trying to achieve the column selection?
Do you need the column (permamently) highlighted as a visual feature in itself?
Do you need the values to be selected and copied?
What kind of behavior specifically are your use cases requiring?

**Comment by lhotamir** [comment]
I am in need of a simillar feature. I would like to highlight whole column when hovering over individual cells like a row is highlighted.

**Comment by alfonsoar** [comment]
@joserodolfofreitas 

> Could you expand on why you are trying to achieve the column selection?
I can't go into to much detail on what the exact use case is but the overall usecase is that we need the ability to "select a column" get some metadata from it and then do some operations with that data. 


> Do you only need the column (permamently) highlighted as a visual feature in itself?
Ideally no, but I can get the metadata I need by doing some manipulation on the `rowData` I pass in. 


> Do you need the values to be selected and copied?
Selected would be nice, we don't have a usecase for copied at this time. 


> What kind of behavior specifically do your use cases require?
The behavior we would ideally like to see is: 
1. When a user selects a column we are able to get the data that is selected. 
2. When a user selects a column we are able to see the "1 column" selected in the data grid. 
3. A "controlled" prop for selecting the column would also be nice. We sometimes have the usecase of persisting the column selection until another action happens.

**Comment by nhaebigkerber** [comment]
Hi, my organization is very interested in this. I've implemented a custom solution using sx injection and cell classNames, but it's not elegant and it's difficult to modularize. Has there been any progress on this ticket since 2023?
